pkg/forwarder/forwarder_health_test.go

```diff
@@ -55,13 +70,14 @@ func TestHasValidAPIKeyErrors(t *testing.T) {
 	defer ts1.Close()
 	defer ts2.Close()
 
-	keysPerDomains := map[string][]string{
+	keysPerAPIEndpoint := map[string][]string{
```

Comment:
Unless I'm missing something, this test shouldn't need to be changed, it should pass as-is. Is there a reason you modified it?

Reply:
In HasValidAPIKey, I updated the domain,keys for loop to iterate over fh.keysPerAPIDomain rather than fh.keysPerDomain so that the API domains are used in hasValidAPIKey() func, which forced this test to fail unless I updated it with the keysPerAPIDomain map.  This made the test pass again.  I'm open to changing the way I call this, but am not sure how else I might get the api validation loop to evaluate the proper endpoints.

Reply:
the test calls `fh.init()`, which should populate `fh.keysPerAPIDomain` from `fh.keysPerDomains`. So I'm pretty sure the test should pass as it was before this PR, can you give it a try?